Update GMSA docs for beta in v1.16 (#16579)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-gmsa.md

{{% capture overview %}}

{{< feature-state for_k8s_version="v1.16" state="beta" >}}

This page shows how to configure [Group Managed Service Accounts](https://docs.microsoft.com/en-us/windows-server/security/group-managed-service-accounts/group-managed-service-accounts-overview) (GMSA) for pods and containers that will run on Windows nodes. Group Managed Service Accounts are a specific type of Active Directory account that provides automatic password management, simplified service principal name (SPN) management, and the ability to delegate the management to other administrators across multiple servers.

> In Kubernetes, GMSA credential specs are configured at a Kubernetes cluster-wide scope as custom resources. Windows pods, as well as individual containers within a pod, can be configured to use a GMSA for domain based functions (e.g. Kerberos authentication) when interacting with other Windows services. As of v1.16, the Docker runtime supports GMSA for Windows workloads.

Reviewer: nits:
line 11, capitalize Pods?
line 13, capitalize Custom Resources, Pods, Pod?

> In Kubernetes, GMSA credential specs are configured at a Kubernetes cluster-wide scope as Custom Resources. Windows Pods, as well as individual containers within a Pod, can be configured to use a GMSA for domain based functions (e.g. Kerberos authentication) when interacting with other Windows services. As of v1.16, the Docker runtime supports GMSA for Windows workloads.

Reviewer: @ddebroy , I read through the page again. A few more nits:
- line 13: To clarify, the last sentence is: As of Kubernetes v1.16, the Docker runtime ...? Or are you referring to the Docker version here?
- line 22: change the docs link to the feature gates page (minus the backticks):
`See [Feature Gates](/docs/reference/command-line-tools-reference/feature-gates/) for an explanation of enabling or disabling feature gates.`
- line 25: change the docs link to the CRD page:
`A [CustomResourceDefinition](/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/)`
- Try to capitalize the word Pod throughout the page. There are just a few places remaining.
- For consistency, change the instances of "use" verb to `use` verb?
- line 204: Should be changed to: If you are having difficulties getting GMSA to work in your environment, there are a few troubleshooting steps you can take:
  OR: If you are having difficulties getting GMSA to work in your environment, you can follow these steps to troubleshoot your setup:

Author: Line 13, the last sentence is indeed `As of Kubernetes v1.16, the Docker runtime`. Not referring to Docker versions.

Addressed the other nits. Thanks for reviewing!

{{% /capture %}}

{{% capture prerequisites %}}

You need to have a Kubernetes cluster and the kubectl command-line tool must be configured to communicate with your cluster. The cluster is expected to have Windows worker nodes where pods with containers running Windows workloads requiring GMSA credentials will get scheduled. This section covers a set of initial steps required once for each cluster:

### WindowsGMSA feature gate

> In the beta state, the `WindowsGMSA` feature gate (required to pass down GMSA credential specs from the pod specs to the container runtime) is enabled by default on the API server and Kubelet. See [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) for an explanation of enabling feature gates in prior releases.

Reviewer 2: nit and tweak:
```suggestion
The `WindowsGMSA` feature gate (required to pass down GMSA credential specs from the pod specs to the container runtime) is enabled by default on the API server and the kubelet. See [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) for an explanation of enabling or disabling feature gates.
```

Author: Thanks - done!

1. The container runtime configures each Windows container with the specified GMSA credential spec so that the container can assume the identity of the GMSA in Active Directory and access services in the domain using that identity.

## Troubleshooting

If you are having difficulties getting gmsa to work in your environment, they are a few troubleshooting steps you can take.

> 

Reviewer: hello @ddebroy . Several nits in this section of the page:
- capitalize GMSA
- capitalize Pod
- remove `we`, lines 206 and 219
In the example below, the Pod ...
If your pod ... then next check communication with the ...
